### PR TITLE
feat: improve AZ logic for addons

### LINKED ISSUE #208: Validate Availability Zone Ordering for `vpc_cni_custom_subnet`s
https://github.com/defenseunicorns/terraform-aws-eks/blob/2b7847ef9e55c347aaaf63376cbe007df62d721a/main.tf#L70

It is possible to supply unordered AZs, that will get `zipmap`ed to subnets in different AZs.

### PR DESCRIPTION
**Pull request by willswire**
closes #208

### CODE REVIEW

**Comment by zack-is-cool** [comment]
pls run
```bash
make pre-commit-all
 ```
or run pre-commit locally to do linty things 👍

File: examples/complete/main.tf [hunk 1]
```diff
@@ -44,7 +44,6 @@ locals {
 ################################################################################
 
 locals {
-  azs              = [for az_name in slice(data.aws_availability_zones.available.names, 0, min(length(data.aws_availability_zones.available.names), var.num_azs)) : az_name]
```
[the remark this answers is not in the document]
Comment (willswire) [reply examples/complete/main.tf]: restored

File: examples/complete/main.tf [hunk 2]
```diff
@@ -63,7 +62,6 @@ module "vpc" {
   name                         = local.vpc_name
   vpc_cidr                     = var.vpc_cidr
   secondary_cidr_blocks        = var.secondary_cidr_blocks
-  azs                          = local.azs
```
[the remark this answers is not in the document]
Comment (willswire) [reply examples/complete/main.tf]: restored

**Review by zack-is-cool — APPROVED** [review]

**Pull request closed without merging**